erpnext/stock/doctype/delivery_trip/delivery_trip.js

```diff
@@ -66,6 +66,52 @@ frappe.ui.form.on('Delivery Trip', {
 		}
 	},
 
+	validate: function (frm) {
+		if (frm.is_new() && frm.doc.delivery_stops) {
+			// frappe.confirm doesn't wait for callback to validate, so creating
+			// Promise to make sure execution is paused until user input
+			return new Promise((resolve, reject) => {
+				frappe.call({
+					method: "erpnext.stock.doctype.delivery_trip.delivery_trip.validate_unique_delivery_notes",
+					args: {
+						delivery_stops: frm.doc.delivery_stops
+					}
+				}).then((r) => {
+					if (r.message.length) {
+						let confirm_message = `Atleast one of the entered Delivery Notes already exist in
+											the following Delivery Trips: ${r.message.join(", ")}.
+											Do you still want to continue?`;
+
+						frappe.confirm(
+							__(confirm_message),
+							() => resolve(),  // If "Yes" is selected
+							() => frappe.set_route("List", frm.doc.doctype)  // If "No" is selected
+						);
+					} else { resolve(); };
```

Comment:
I'm copy/pasting my earlier response before github freaked out and thought the file changed:

The promisified version of this looks great, but you have just a few things to fix.

Always return a value whether you resolve() or reject()
See: https://github.com/xjamundx/eslint-plugin-promise/blob/master/docs/rules/always-return.md

Otherwise you break promise chains, so here on this line you want:

```return resolve();```

Every where else you at least want to return a null:

```return null;```

Also, you want to add a catch to that frappe.call so that you can reject on errors. Otherwise, the promise will eat those without a sign of something going wrong...

```frappe.call(...args...).fail((err) => { return reject(err); });```